environment_switcher/lib/src/environment_switcher.dart

```diff
@@ -0,0 +1,195 @@
+import 'package:flutter/material.dart';
+
+import 'environment.dart';
+import 'environment_store.dart';
+import 'select_environment_sheet.dart';
+
+export 'environment.dart';
+export 'environment_store.dart';
+
+class EnvironmentSwitcher extends StatefulWidget {
+  final Widget Function(Environment) builder;
+  final List<Environment> environments;
+  final Environment defaultEnvironment;
+  final EnvironmentStore environmentStore;
+  final bool showBanner;
+
+  /// A banner that visually shows the user what [Environment]
+  /// is currently set to.
+  /// This is useful because it's possible to be on one Flavour
+  /// in the native code, and another in the Flutter code.
+  const EnvironmentSwitcher({
+    @required this.builder,
+    @required this.environments,
+    this.environmentStore, // mainly required for testing
```

Comment:
Why? How is it used?

Reply:
I'm not entirely sure why, but I assume it's because the Store library tries to access native code, which the testing platform doesn't seem to like. You can try it out by commenting out the `_MockEnvironmentStore` instance in the test. It throws a testing error. Creating a mock environment store seems to avoid this issue.

Reply:
Oh I see! It creates a default version otherwise, and that default uses platform code so can't run in unit tests.